pkg/controller/registry/resolver/resolver.go

```diff
@@ -99,10 +100,11 @@ func (r *SatResolver) SolveOperators(namespaces []string, csvs []*v1alpha1.Clust
 		if current == nil && len(sub.Spec.StartingCSV) > 0 {
 			channelFilter = append(channelFilter, WithCSVName(sub.Spec.StartingCSV))
 			startingCSVs[sub.Spec.StartingCSV] = struct{}{}
+			allowSkipped = true
```

Comment:
instead of a `bool` flag that gets threaded all the way down, could we come up with a `predicate` or `channelFilter` that we could pass to the existing `getSubscriptionInstallables`?

If the resolver only sees the filtered data with one option per channel, the operator will get ignored because it's skipped (because all the operators that skip it will have been filtered out)

Reply:
I agree, this also would help with keeping the signatures of the various helper functions involved in resolution simpler

Reply:
I have tried to do so but the issue here is being-skipped is not a property that can be constructed into `predicate`. You can filter out an operator with a property that owns (or doesn't). Currently you can filter out an operator that skips/replaces something but you can't do the other way around.
At the moment, we filter out the skipped version during sorting bundles within a channel via a utility function. In order to filter out skipped versions, we will need to let the registry to add some label manually when the bundle is added to an index.

Reply:
I don't think we need to filter out "things that skip startingCSV"

I think we need to filter out "anything that is not startingCSV" - if you've specified startingCSV and there's not already an operator installed for the subscription, you should either get startingCSV or nothing, right?

It might be as simple as changing

```go
channelFilter = append(channelFilter, WithCSVName(sub.Spec.StartingCSV))
```
to 
```go
predicates =  append(predicates, WithCSVName(sub.Spec.StartingCSV))
```

(but I haven't looked at this in a while - I'm a little unclear about why the channelFilter doesn't take care of this already)

Reply:
After discussed with Evan, we agreed to proceed with the `predicate` addition to `cachepredicate` to filter out the ineligible versions in advance before it reaches the `sortChannel` stage.